### PR TITLE
migration to v10

### PR DESCRIPTION
**Pull request by guewen**
This is the one-to-one migration of the connector. More ambitious refactorings will follow in other pull requests (#212, ..).

### CODE REVIEW

File: connector/backend_model.py [hunk 1]
```diff
@@ -19,7 +19,7 @@
 #
 ##############################################################################
```
Comment (lasley) [inline connector/backend_model.py]: Please use new, shorter header

File: connector/connector.py [hunk 2]
```diff
@@ -351,14 +349,14 @@ class Binder(ConnectorUnit):
...
     _external_field = 'external_id'  # override in sub-classes
     _backend_field = 'backend_id'  # override in sub-classes
-    _openerp_field = 'openerp_id'  # override in sub-classes
+    _odoo_field = 'odoo_id'  # override in sub-classes
     _sync_date_field = 'sync_date'  # override in sub-classes
 
-    def to_openerp(self, external_id, unwrap=False):
-        """ Give the OpenERP ID for an external ID
+    def to_odoo(self, external_id, unwrap=False):
+        """ Give the ODoo ID for an external ID
```
Comment (lasley) [inline connector/connector.py]: s/ODoo/Odoo

File: connector/doc/index.rst [hunk 3]
```diff
@@ -51,7 +51,7 @@ Connectors based on the framework
 * `Magento Connector <http://www.odoo-magento-connector.com>`_
 * `Prestashop Connector <https://github.com/OCA/connector-prestashop>`_
 * `solerp (Solr Connector) <https://github.com/akretion/solerp>`_
-* `Odoo Multi Company <http://www.openerp.net.cn/new-intercompany-process-module/>`_
+* `Odoo Multi Company <http://www.odoo.net.cn/new-intercompany-process-module/>`_
```
Comment (lasley) [inline connector/doc/index.rst]: This URI is (and previously was) invalid. Is this it - https://github.com/OCA/connector-odoo2odoo ?

File: connector/queue/queue_data.xml [hunk 4]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<openerp>
+<odoo>
```
Comment (lasley) [inline connector/queue/queue_data.xml]: `<odoo noupdate="1">` & can remove the data tag

**Review by lasley — APPROVED** [review]
Minor nitpicks, code no test 👍 

Side note - the producer is very sexy now w/ the use of the BaseModel

File: connector/queue/queue_data.xml (hunk 4 above)
Comment (guewen) [reply connector/queue/queue_data.xml]: I had to leave it because there is one part with `noupdate="1"` and another one with `noupdate="0"` in the file.

**Comment by lasley** [comment]
FYI functional test on this looks good too

**Comment by guewen** [comment]
This PR still includes the queue, we'll have to continue the work on #212 which was already split in `connector` and `queue_job` (but the latter has to be dropped as moved in its own repo).

**Comment by lasley** [comment]
Thanks for clarifying @guewen - I was kind of curious whether we would be refactoring for the new queue.

It seems to me like we should still continue forward with the merge of this one though, as it works in its entirety & we are starting to receive other PRs doing the same upgrade. 

Stripping out the queue/jobs should be fairly transparent in terms of the external API, or am I incorrect?

**Comment by lmignon** [comment]
IMO we should wait for the move of the queue to it's own repo since this change is not transparent in terms of the external API.

**Comment by lasley** [comment]
Thanks for the clarification @lmignon

**Comment by bealdav** [comment]
HI Guewen, 
Your are not afraid by a new rename ?
`s/odoo_id/erp_id` can save you for eternity ;-)

**Comment by guewen** [comment]
@bealdav that's an ERP they say.

Anyway, no, I'm not afraid by a new rename, odoo is spread all over the application (xml, python files, ...) so if it had to change again, renaming a few more occurrences would not be so much work.

**Review by mocak — APPROVED** [review]

**Comment by gurneyalex** [comment]
Travis is read because of https://github.com/OCA/odoo-sphinx-autodoc/issues/6 I think

**Comment by gurneyalex** [comment]
OCA/odoo-sphinx-autodoc#7 should fix the build.

**Comment by guewen** [comment]
This PR still need some work, it still contains the queue job that has now been extracted to https://github.com/OCA/queue

**Comment by guewen** [comment]
FYI tests are green travis is red only because of the docs build, which should be better once we have a new release of odoo-sphinx-autodoc (https://github.com/OCA/odoo-sphinx-autodoc/pull/7).

I'll now battle-test a little with a real implementation.

Docs and examples will need to be updated but that can wait.

**Comment by lmignon** [comment]
Thank you @guewen for all this work!

File: connector/__manifest__.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+             ],
+ 'data': ['security/connector_security.xml',
+          'security/ir.model.access.csv',
+          'checkpoint/checkpoint_view.xml',
+          'connector_menu.xml',
+          'setting_view.xml',
+          'res_partner_view.xml',
+          ],
+ 'installable': True,
+ 'application': True,
```
Comment (lmignon) [inline connector/__manifest__.py]: IMO it's not really an application since it doesn't provide functionality alone...

**Comment by guewen** [comment]
I should also take the last commits from 9.0

File: connector/__manifest__.py (hunk 5 above)
Comment (guewen) [reply connector/__manifest__.py]: right, will change that

**Comment by guewen** [comment]
In the last commit, I remove the `env` of the `ConnectorUnit` as I think we should use the one of the backend_record.

File: connector/doc/guides/bootstrap_connector.rst [hunk 6]
```diff
@@ -172,22 +172,20 @@ Environment
...
-        lang = backend_record.default_lang_id
-        lang_code = lang.code if lang else 'en_US'
-        if lang_code == session.context.get('lang'):
-            return env
-        else:
-            with env.session.change_context(lang=lang_code):
-                return env
+      @contextmanager
+      @api.multi
+      def get_environment(self, model_name):
```
Comment (yvaucher) [inline connector/doc/guides/bootstrap_connector.rst:186]: indentation is 6 instead of 8

Comment (guewen) [reply connector/doc/guides/bootstrap_connector.rst:186]: Doc is outdated and need to be refreshed but not in this PR.

Comment (guewen) [reply connector/doc/guides/bootstrap_connector.rst:186]: And: yes we could add `get_environment` in the abstract `connector.backend`. Yes the `model_name` is required for the `ConnectorEnvironment`.

**Comment by LeartS** [comment]
Hi guys, what's the status of this PR? Can it be merged? I'm using the connector module in this PR to build a connector with a SaaS search engine and it seems to work perfectly.

**Review by LeartS — APPROVED** [review]
👍 Quick code review + used to build a specific connector module on top of it, seems to work great.

**Comment by guewen** [comment]
Docs build is now green as well, ready to be merged.

File: connector/doc/index.rst (hunk 3 above)
Comment (lmignon) [reply connector/doc/index.rst]: @guewen Can you remove this URI. Indeed it's invalid.

**Review by lmignon — APPROVED** [review]
@guewen Thank you for all this work! Can you remove the invalid URI into connector/doc/inddex.rst? Once done I'll merge this PR

**Comment by guewen** [comment]
@lmignon done

**Review by lmignon — APPROVED** [review]

**Comment by lmignon** [comment]
@guewen  thank you

**Pull request merged**